src/main/java/org/sagebionetworks/web/client/widget/entity/file/FileDownloadMenuItemViewImpl.java

```diff
@@ -56,9 +53,8 @@ private void clearClickHandlers() {
 	}
 
 	@Override
-	public void setIsAuthorizedDirectDownloadLink() {
-		clearClickHandlers();
-		downloadLinkHandlerRegistration = downloadLink.addClickHandler(authorizedDirectDownloadClickHandler);
+	public void showDownloadError(){
+		downloadLinkHandlerRegistration = downloadLink.addClickHandler(errorDownloadClickHandler);
```

Comment:
I think you still need to call clearClickHandlers() before addClickHandler, since you can add multiple click handlers to the link (which is why we hold onto the downloadLinkHandlerRegistration).